### PR TITLE
TF-1822 save string to file in internal storage mobile

### PR DESCRIPTION
**Pull request by nqhhdev**
- Issue: #1822 

- Resolved : 

Screenshot 2023-05-08 at 23 32 07

### CODE REVIEW

File: core/lib/utils/file_utils.dart [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  static Future<String> _getInternalStorageDirPath(
+    String nameFile,
+    {
+      String? folderPath,
+      String? extensionFile
+    }
+  ) async {
+    if (!BuildUtils.isWeb) {
+
+      String fileDirectory = (await getApplicationDocumentsDirectory()).absolute.path;
```
Comment (hoangdat) [inline core/lib/utils/file_utils.dart:18]: please make sure `getApplicationDocumentsDirectory` is not deprecated in Android version >= 30

File: core/lib/utils/file_utils.dart [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        fileDirectory = '$fileDirectory.$extensionFile';
+      }
+
+      return fileDirectory;
+    } else {
+      throw DeviceNotSupportedException();
+    }
+  }
+
+  static Future<File> saveToFile(
```
Comment (hoangdat) [inline core/lib/utils/file_utils.dart:36]: why we need it as a static func?

**Review by hoangdat — COMMENTED** [review]
please demo this utils with emulator and show the file in ApplicationDocumentDirectory.

File: core/lib/utils/file_utils.dart [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  ) async {
+    String emailContent = '';
+
+    final internalStorageDirPath = await _getInternalStorageDirPath(
+      nameFile,
+      folderPath: folderPath,
+      extensionFile: extensionFile
+    );
+
+    final file = File('$internalStorageDirPath.txt');
```
Comment (dab246) [inline core/lib/utils/file_utils.dart:71]: It is recommended to leave the file with the extension changed. Not `.txt`

File: core/lib/utils/file_utils.dart [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  static Future<String> getFromFile(
+    String nameFile,
+    {
+      String? folderPath,
+      String? extensionFile
+    }
+  ) async {
+    String emailContent = '';
```
Comment (dab246) [inline core/lib/utils/file_utils.dart:63]: IMO, No need to create variables first. Should be written in functional programming

File: core/pubspec.yaml [hunk 5]
```diff
@@ -63,6 +63,12 @@ dependencies:
 
   http_parser: 4.0.2
 
+  path_provider: 2.0.13
+
+  external_path: 1.0.3
```
Comment (dab246) [inline core/pubspec.yaml:68]: IMO, Just using `path_provider` is enough

**Review by dab246 — APPROVED** [review]

**Pull request merged**